lib/marc/datafield.rb

```diff
@@ -123,52 +141,34 @@ def append(subfield)
       @subfields.push(subfield)
     end
 
-    
-
-    # You can iterate through the subfields in a Field:
-    #   field.each {|s| print s}
-
-    def each
-      for subfield in subfields
-        yield subfield
-      end
-    end
-
-    #def each_by_code(filter)
-    #  @subfields.each_by_code(filter)
-    #end
-
-    # You can lookup subfields with this shorthand. Note it 
+    # You can lookup subfields with this shorthand. Note it
     # will return a string and not a MARC::Subfield object.
+    #
+    # NOTE: the value of only the *first* subfield is returned
     #   subfield = field['a']
-    
+
     def [](code)
-      subfield = self.find {|s| s.code == code}
+      subfield = self.detect {|s| s.code == code}
       return subfield.value if subfield
-      return
+      return nil
     end
```

Comment:
Maybe the result here could be `subfield.value || nil` or, if that might raise an exception, `subfield.value rescue nil`